index.js

```diff
@@ -1,114 +1,138 @@
-var output = argv._[1] || argv.o || argv.output
+var HTML_DPI = 96
 
-if (argv.v || argv.version) {
-  console.log('v' + pkg.version)
-  process.exit(0)
-}
+var PDFExporter = function (input, output, argv) {
+  app.on('ready', appReady)
+  app.on('window-all-closed', function () {
+    if (process.platform !== 'darwin') {
+      app.quit()
+    }
+  })
 
-if (argv.h || argv.help) {
-  usage(1)
-} else if (!input || !output) {
-  usage(1)
-}
+  function appReady () {
+    var customCss = argv.css
+
+    function isMarkdown (input) {
+      var ext = path.extname(input)
+      return ext.indexOf('md') > 0 || ext.indexOf('markdown') > 0
+    }
 
-app.on('ready', appReady)
+    if (isMarkdown(input)) {
+      var opts = {}
 
-app.on('window-all-closed', function () {
-  if (process.platform !== 'darwin') {
-    app.quit()
-  }
-})
+      if (customCss) {
+        opts.customCss = customCss
+      }
 
-function appReady () {
-  var customCss = argv.c || argv.css
+      // if given a markdown, render it into HTML and return the path of the
+      // HTML
+      input = markdownToHTMLPath(input, opts, function (err, tmpHTMLPath) {
+        if (err) {
+          console.error('Parse markdown file error', err)
+          app.quit()
+        }
 
-  function isMarkdown (input) {
-    var ext = path.extname(input)
-    return ext.indexOf('md') > 0 || ext.indexOf('markdown') > 0
+        var indexUrl = wargs.urlWithArgs(tmpHTMLPath, {})
+        render(indexUrl, output)
+      })
+    } else {
+      var indexUrl = wargs.urlWithArgs(input, {})
+      render(indexUrl, output)
+    }
   }
 
-  if (isMarkdown(input)) {
-    var opts = {}
+  /**
+   * render file to pdf
+   * @param  {String} indexUrl The path to the HTML or url
+   */
+  function render (indexUrl, output) {
+    var png = output.toLowerCase().endsWith('.png')
+
+    var pdfOptions = {
+      marginsType: argv.marginsType,
+      printBackground: argv.printBackground,
+      printSelectionOnly: argv.printSelectionOnly,
+      pageSize: argv.pageSize,
+      landscape: argv.landscape
+    }
 
-    if (customCss) {
-      opts.customCss = customCss
+    var pageDimensions = {
+      'A4': {x: 595, y: 842},
+      'Letter': {x: 8.5 * HTML_DPI, y: 11 * HTML_DPI},
+      'Legal': {x: 8.5 * HTML_DPI, y: 14 * HTML_DPI}
     }
 
-    // if given a markdown, render it into HTML and return the path of the HTML
-    input = markdownToHTMLPath(input, opts, function (err, tmpHTMLPath) {
-      if (err) {
-        console.error('Parse markdown file error', err)
-        app.quit()
+    var pageDim = pageDimensions[pdfOptions.pageSize]
+    pageDim = pdfOptions.landscape ? {x: pageDim.y, y: pageDim.x} : pageDim
+    var winX = argv.windowX || pageDim.x
+    var winY = argv.windowY || pageDim.y
+
+    console.info('Opening a browser window', winX, 'x', winY)
+    var win = new BrowserWindow({
+      width: winX,
+      height: winY,
+      x: 0, y: 0,
+      show: true,
```

Comment:
Some of these options need cleaned up before you merge.  Accepting a hash of browser options is probably the way to go here because there are so many.  I saw a similar PR I believe.